pkg/queryfrontend/queryrange.proto

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+syntax = "proto3";
+
+package queryfrontend;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "store/storepb/types.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+message ThanosRequest {
```

Comment:
Hmmmmmmm sad we need to do it. Can we reuse SeriesRequest instead mabye? :thinking:

Reply:
What's the benefit of reusing it? SeriesRequest has some unnecessary fields like `aggregates` and `skipt_chunks` and `hints`. Also we also need `replicaLabels`, `dedup` fields in `ThanosRequest`. What do you think?

Reply:
Well it allows to reduce number of protobuf code we have, and the code itself is non trivial.

And I don't think extra fields are the problem, just rather the fact that it's different level -> matchers vs query.

And the benefit of reuse is quite important. You change request for StoreAPI - most likely this one has to change as well.

Why do we need to have this protobuf BTW? It's all HTTP API? I believe if we want to suddenly have all HTTP API coded in protobuf (which nice idea BTW), probably we need to do it properly. Name it query API and make sure we can generate docs, and we have same structs for other HTTP APIs  (btw we have already LabelRequest, etc)

So after Storepb and this is really really similar if not the same (minus,plus few fields).

Reply:
Thanks, you are right. It is not necessary to have a protobuf for it. Using one struct is enough. Let me update it.